src/healthData/HealthDataUtils.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2015 Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, maxerr: 50 */
+/*global define, $ */
+
+define(function (require, exports, module) {
+    "use strict";
+    
+    var ExtensionManager = require("extensibility/ExtensionManager");
+    
+    /**
+     * Get the OS Version
+     * @return {String} version of OS 
+     */
+    function getOSVersion() {
+        var winOSStrings = {
+            "5.0" : "Windows 2000",
+            "5.1" : "Windows XP",
+            "5.2" : "Windows XP | Windows Server 2003 | Windows Home Server",
+            "6.0" : "Windows Vista | Windows Server 2008",
+            "6.1" : "Windows 7 | Widnows Server 2008 | Widnows Home Server 2011",
+            "6.2" : "Windows 8 | Windows Server 2012",
+            "6.3" : "Windows 8.1 | Windows Server 2012 R2",
+            "10.0" : "Windows 10"
+        };
+        
+        var windowRegex = /Windows NT (\d+\.\d+)/i,
+            macRegex    = /Macintosh;\s([\w\W]+?)(\)|;)[\w\W]+/i,
+            linuxRegex  = /X11;\s([\w\W]+?)(\)|;)[\w\W]+/i;
+        
+        var userAgent = navigator.userAgent;
+        var osVersion = userAgent;
+        
+        if (/Windows/.test(userAgent)) {
+            osVersion = winOSStrings[windowRegex.exec(userAgent)[1]] || osVersion;
```

Comment:
I'm not a huge fan of trying to prettify the version strings here on the client side. It's unnecessary and could make data analysis trickier later.  Examples:

* Let's say we condense all versions of Windows "6.3" into the single result string "Windows 8.1 | Windows Server 2012 R2".  Later, MS releases a Windows 8.1 R2 which has "6.3.1" as its version instead. Now we can't distinguish between the two.
* Let's say a new Windows 11 comes out in the future.  If we don't update Brackets early enough, half the data will have the full userAgent as its version (version not found in `winOSStrings`) and half the data will have "Windows 11" as its version (after we update Brackets, now it's recognized).  We'll need special logic on the server to combine the two buckets.
* Same scenario could play out if a CEF update changes the userAgent format on any of the OSes we support, but we don't realize it until after shipping.

So it feels better to either send the raw userAgent (which there's tons of 3rd party libraries we can use to analyze on the server later) or to at least just send the result of the regexp match without any mapping to something else.